Modules/rats.py

```diff
@@ -0,0 +1,183 @@
+"""
+rats.py - Rats object
+
+Handles the rats cache and provides facilities for managing rats.
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+"""
+
+import logging
+from uuid import UUID
+
+import config
+from ratlib.names import Platforms
+
+LOG = logging.getLogger(f"{config.Logging.base_logger}.{__name__}")
+
+
+class Rats(object):
+    """
+    This class keeps track of known rats as they are used and stores them in a
+    class cache.
+
+    Instances of this class are used to represent a unique, individual rat.
+
+    Creation of a `Rats` object will automatically add the created rat to the
+    cache, allowing convenience method `Rats.get_rat` to return the instance
+    when called.
+    """
+
+    cache_by_id = {}
+    """Cache of rat objects by their UUID"""
+    cache_by_name = {}
+    """Cache of rat objects by rat name (str)"""
+
+    def __init__(self, uuid: UUID, name: str = None, platform: Platforms = Platforms.DEFAULT):
+        """
+        Creates a new rat
+
+        Args:
+            uuid (UUID):
+            name (str): rat's name
+            platform (Platforms): rat's platform
+        """
+        # set our properties
+        self._platform = platform
+        self._uuid = uuid
+        self._name = name
+        # and update the cache
+        if name and name not in Rats.cache_by_name:
+            # don't register duplicates
+            Rats.cache_by_name[name] = self
+        if uuid not in Rats.cache_by_id:
+            # don't register duplicates
+            Rats.cache_by_id[uuid] = self
+
+    @property
+    def uuid(self):
+        """
+        API id of the rat
+
+        Returns:
+            UUID
+        """
+        return self._uuid
+
+    @uuid.setter
+    def uuid(self, value) -> None:
+        """
+        Set the UUID of the rat.
+        accepts a uuid encoded string, or a uuid object.
+
+        Args:
+            value (UUID): new uuid for rat
+
+        Returns:
+            None
+        """
+        if isinstance(value, str):
+            LOG.debug(f"value was a string with data '{value}'")
+            uuid = UUID(value)
+            LOG.debug("parsed value into a valid UUID.")
+            self._uuid = uuid
+        elif isinstance(value, UUID):
+            self._uuid = value
+
+        else:
+            raise TypeError(f"expected type None or type str, got "
+                            f"{type(value)}")
```

Comment:
The string representation for types is `<class 'int'>` etc.. You'll want to use `type(value).__name__` here.

Reply:
calling a magic method for a type error does not feel like an elegant solution. The fact an exception is being raised here and the information provided by the string representation should be sufficient for debugging purposes

Reply:
All good, this is, of course, your decision and only a cosmetic change.
Incidentally, __name__ is not a method but an attribute (or property, not actually sure) and since no function exists for getting the name of a class, this is generally how it's done.